.github/workflows/ci-cd.yml

```
@@ -0,0 +1,527 @@
+
+      - name: Run E2E tests
+        run: npm run test:e2e:run
+        env:
+          NODE_ENV: test
+          TEST_DB_HOST: localhost
+          TEST_DB_PORT: 5433
+          TEST_REDIS_HOST: localhost
+          TEST_REDIS_PORT: 6380
+
+      - name: Upload E2E test reports
+        uses: actions/upload-artifact@v3
```

actionlint (1.7.7)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)